testing/utils.go

```diff
@@ -33,3 +36,19 @@ func GenerateString(length uint) string {
 	}
 	return string(bytes)
 }
+
+// Parse out msg responses from a transaction result
+func UnmarshalMsgResponse(cdc *codec.LegacyAmino, resp abci.ExecTxResult, msgs ...codec.ProtoMarshaler) error {
+	var txMsgData sdk.TxMsgData
+	if err := cdc.Unmarshal(resp.Data, &txMsgData); err != nil {
+		return err
+	}
+
+	for i, msg := range msgs {
+		if err := cdc.Unmarshal(txMsgData.MsgResponses[i].Value, msg); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
```

Comment:
thanks @phamminh0811! 

It looks like we have some very similar code in our [e2es](https://github.com/cosmos/ibc-go/blob/08d36eb0735caa29ddf981a872d0a55d218be3f6/e2e/testsuite/codec.go#L96-L120) at the moment, I wonder if we could maybe add the functionality here, and also refactor the E2E version to call into this, WDYT?

Reply:
agree!

Reply:
got it, I'm gonna create a new pr and refactor that